### PR TITLE
remove the `new_method` kwarg and deprecation warning.

### PR DESCRIPTION
**Pull request by angadsinghsandhu**
#### References to other Issues or PRs
Fixes (#7790), specifically the review left by @moorepants , at this link : [Click Here](https://github.com/sympy/sympy/pull/7790#discussion_r15885290)

#### Brief description of what is fixed or changed
This is a simple patch referring to the TODO given on line `476` in `sympy/sympy/physics/mechanics/kane.py` i.e. : https://github.com/sympy/sympy/blob/0f782091845fbb52b26c3d55fb50240c260054b4/sympy/physics/mechanics/kane.py#L476

The depreciated `new_method` parameter has been removed and the depreciation warning has also been removed.

#### Release Notes

* physics.mechanics
  * removed the new_method kwarg and deprecation warning

### CODE REVIEW

File: sympy/physics/mechanics/kane.py [hunk 1]
```diff
@@ -473,8 +473,7 @@ def to_linearizer(self):
         return Linearizer(f_0, f_1, f_2, f_3, f_4, f_c, f_v, f_a, q, u, q_i,
                 q_d, u_i, u_d, r)
 
-    # TODO : Remove `new_method` after 1.1 has been released.
-    def linearize(self, *, new_method=None, **kwargs):
+    def linearize(self, *, **kwargs):
```
Comment (sidhu1012) [inline sympy/physics/mechanics/kane.py]: It should be  ``` def linearize(self, **kwargs)```

**Comment by moorepants** [comment]
Some relevant issues:

https://github.com/sympy/sympy/issues/11199

https://github.com/sympy/sympy/issues/9641

One major issue is that the new method failed to solve the test mentioned in #9641. Secondly, this PR doesn't actually remove the old method or the deprecation warning or leave the `new_method` kwarg in place so that code doesn't break.

**Comment by angadsinghsandhu** [comment]
@moorepants the `physics/mechanics/examples/lin_pend_nonmin_example.rst` is still using the _"new_method"_ parameter in the linearize method. Should that be removed from the `.rst` file as well?

But, the `**kwargs` should still be able to accept _"new_method"_ if it is able to accept the _"op_point"_ and _"A_and_B"_ arguments. Could this issue be related to putting the _"new_method"_ in a new line in `lin_pend_nonmin_example.rst` ?

**Comment by moorepants** [comment]
I don't know. You should start by explaining what was actually deprecated and what changes are needed to finalize the deprecation. We can remove things that have been deprecated for some releases. But there is also the issue that the new method doesn't work for the test the one issue! That's a much bigger thing to address.

**Comment by angadsinghsandhu** [comment]
> I don't know. You should start by explaining what was actually deprecated and what changes are needed to finalize the deprecation. We can remove things that have been deprecated for some releases. But there is also the issue that the new method doesn't work for the test the one issue! That's a much bigger thing to address.

@moorepants It seems the exact same linearize function was declared until `v1.6.2` [(Click Here for Reference)](https://github.com/sympy/sympy/blob/c3087c883be02ad228820ff714ad6eb9af1ad868/sympy/physics/mechanics/kane.py#L476).

As defined in `v1.6.2`:
```py
def linearize(self, **kwargs):
```

As defined in `v1.7.1`:
```py
def linearize(self, *, new_method=None, **kwargs):
```

As defined in this PR:
```py
def linearize(self, **kwargs):
```

I definitely think the issue might be in `lin_pend_nonmin_example.rst`. I will make some changes to it and commit to see if all the checks are passed. I will hard rest and show my findings then.

**Pull request closed without merging**